tests/Unit/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/Test_DampedHarmonicGaugeQuantities.cpp

```diff
@@ -0,0 +1,1338 @@
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    d4_shift.get(0, 1 + i) = dt_shift.get(i);
+    for (size_t j = 0; j < SpatialDim; ++j) {
+      d4_shift.get(1 + j, 1 + i) = d_shift.get(j, i);
+    }
+  }
+
+  // \f[ \partial_a (\mu_S/N) = (1/N) \partial_a \mu_{S}
+  //         - (\mu_{S}/N^2) \partial_a N
+  // \f]
+  auto d4_muS_over_N =
+      make_with_value<tnsr::a<DataVector, SpatialDim, Frame::Inertial>>(
+          get(lapse), 0.);
+  {
+    auto prefac = -mu_S * one_over_lapse * one_over_lapse;
+    for (size_t a = 0; a < SpatialDim + 1; ++a) {
+      d4_muS_over_N.get(a) +=
+          one_over_lapse * d4_mu_S.get(a) + prefac * d4_N.get(a);
+    }
+  }
+
+  // Calc \f$ \partial_a T3 \f$ (note minus sign)
+  auto dT3 =
+      make_with_value<tnsr::ab<DataVector, SpatialDim, Frame::Inertial>>(x, 0.);
+  for (size_t a = 0; a < SpatialDim + 1; ++a) {
+    for (size_t b = 0; b < SpatialDim + 1; ++b) {
+      for (size_t i = 0; i < SpatialDim; ++i) {
+        dT3.get(a, b) -= d4_muS_over_N.get(a) * spacetime_metric.get(b, i + 1) *
+                         shift.get(i);
+        dT3.get(a, b) -=
+            mu_S_over_N * d4_spacetime_metric.get(a, b, i + 1) * shift.get(i);
+        dT3.get(a, b) -= mu_S_over_N * spacetime_metric.get(b, i + 1) *
+                         d4_shift.get(a, i + 1);
+      }
+    }
+  }
+
+  // Initialize initial gauge function and its roll-off function
+  const auto gauge_h_init = make_with_value<typename db::item_type<
+      GeneralizedHarmonic::Tags::InitialGaugeH<SpatialDim, Frame::Inertial>>>(
+      x, 0.);
+  const auto d4_gauge_h_init = make_with_value<typename db::item_type<
+      GeneralizedHarmonic::Tags::SpacetimeDerivInitialGaugeH<SpatialDim,
+                                                             Frame::Inertial>>>(
+      x, 0.);
+
+  // Calc \f$ \partial_a H_b = dT3_{ab} \f$
+  const auto& d4_gauge_h_expected = dT3;
+
+  // Check that locally computed \f$\partial_a H_b\f$ matches returned one
+  typename db::item_type<GeneralizedHarmonic::Tags::SpacetimeDerivGaugeH<
+      SpatialDim, Frame::Inertial>>
+      d4_gauge_h{};
+  GeneralizedHarmonic::compute_spacetime_deriv_damped_harmonic_h<
+      SpatialDim, Frame::Inertial>(
+      make_not_null(&d4_gauge_h), gauge_h_init, d4_gauge_h_init, lapse, shift,
+      spacetime_unit_normal_one_form, sqrt_det_spatial_metric,
+      inverse_spatial_metric, spacetime_metric, pi, phi, t, x, 0., 0.,
+      amp_coef_S, 0, 0, exp_S, /* exponents */
+      // roll on function parameters for lapse / shift terms
+      -1., -1., -1., -1., -1., -1., t_start_S, sigma_t_S,
+      r_max /* weight function */);
+
+  CHECK_ITERABLE_APPROX(d4_gauge_h_expected, d4_gauge_h);
+}
+}  // namespace
+
+SPECTRE_TEST_CASE(
+    "Unit.Evolution.Systems.GeneralizedHarmonic.GaugeSource.DampedHarmonic",
+    "[Unit][Evolution]") {
+  pypp::SetupLocalPythonEnvironment local_python_env{
+      "Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/"};
```

Comment:
This you can make `""` to be able to import `from PointwiseFunctions`

Reply:
done